Makefile

```diff
@@ -530,28 +522,13 @@ $(TELEKUBE_OUT): packages
 		--skip-version-check \
 		-o $(TELEKUBE_OUT)
 
-$(GRAVITY_OUT):
-	$(MAKE) -C build.assets build BINARIES=gravity
-
-$(TELE_OUT):
-	$(MAKE) -C build.assets build BINARIES=tele
-
-#
-# builds robotest installer
-#
-.PHONY: robotest-image
-robotest-image: GRAVITY=$(GRAVITY_OUT) --state-dir=$(PACKAGES_DIR)
-robotest-image: $(ROBOTEST_OUT)
-
-ROBOTEST_TAR_SRC=$(shell find $(ASSETSDIR)/robotest/ -type f)
+.PHONY: robotest-run-suite
+robotest-run-suite: # depends on: telekube opscenter $(TELE_OUT) $(GRAVITY_OUT)
```

Comment:
Yes.

These .`.PHONY` upstream targets don't track dependency relations between sources and artifacts, meaning if `robotest-run-suite` depends on them directly it will always rebuild `production telekube opscenter`. This is tedious for anything but an automated builder that does it from scratch each time.  `robotest-run-suite` used to not declare any dependencies (which is a lie) -- and these commented out dependencies are my compromise.

We might similarly ask why all the `packages` targets don't depend on `$(GRAVITY_OUT)`? Or why doesn't  `$(TELE_OUT)` doesn't depend on `$(TELE_OUT)`?  They should, but to do that we need proper dependency relations between `$(GRAVITY_OUT)` and `$(TELE_OUT)` and their upstreams.  This quickly balloons into fixing the entire build.

I felt fixing latent broken dependency relationships  was out of scope. Especially as make's timestamp based tracking leaves a lot to be desired.  If we're going to invest this sort of energy into the build, it should be coupled with a modern build system IMO.

Reply:
Agree - we should better focus on mage progress instead of fixing makefiles.